ci/local-run.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u
+
+function check_var() {
+  [[ ${!1-} ]] && return 0 || true
+  [[ -f .env ]] || { echo "Var $1 missing from environment and no .env file found; see .env.sample file"; return 1; }
+  grep -q "^$1=" .env || { echo "Var $1 missing both from environment and .env file"; return 1; }
+}
+
+image_name=$1
+
+if [[ $OSTYPE == 'darwin'* ]]; then  # group id 20 is usually taken in MacOS
+  group_ID="500"
+else
+  group_ID="$(id -g)"
+fi
+
+cd "$(git rev-parse --show-toplevel)"/ci/$image_name/
+if [[ "$image_name" == "tox" ]]; then
```

Comment:
I'd rather avoid `if`s like that, that generally leads to spaghetti code pretty quickly 🍝
Instead, I'd extract some generic scripts/functions that can be reused across the `local-run.sh` scripts (I think at least deriving `DIRECTORY_HASH` and running `docker-compose build` are good candidates)

Reply:
(I'd opt for shell functions here, these things seem to trivial to merit separate scripts TBH)